src/codegen/doc/format.rs

```diff
@@ -191,35 +191,45 @@ fn find_type(name: &str, env: &Env) -> String {
         .find(|(_, o)| o.c_type == name)
         .map(|(_, o)| o)
     {
-        symbols.by_tid(obj.type_id)
+        let sym = symbols.by_tid(obj.type_id);
+        sym.map(|sym| format!("[`{name}`][crate::{name}]", name = sym.full_rust_name()))
     } else if let Some(record) = env
         .analysis
         .records
         .iter()
         .find(|(_, r)| r.type_(&env.library).c_type == name)
         .map(|(_, r)| r)
     {
-        symbols.by_tid(record.type_id)
-    } else if let Some(enum_) = env
+        let sym = symbols.by_tid(record.type_id);
+        sym.map(|sym| format!("[`{name}`][crate::{name}]", name = sym.full_rust_name()))
+    } else if let Some(enum_info) = env
         .analysis
         .enumerations
         .iter()
         .find(|e| e.type_(&env.library).c_type == name)
     {
-        symbols.by_tid(enum_.type_id)
-    } else if let Some(flag) = env
+        let sym = symbols.by_tid(enum_info.type_id).unwrap();
+        Some(format!(
+            "[{name}](crate::{parent})",
+            name = enum_info.name,
+            parent = sym.parent().trim_end_matches("::")
+        ))
+    } else if let Some(flag_info) = env
         .analysis
         .flags
         .iter()
         .find(|f| f.type_(&env.library).c_type == name)
     {
-        symbols.by_tid(flag.type_id)
+        let sym = symbols.by_tid(flag_info.type_id).unwrap();
+        Some(format!(
+            "[{name}](crate::{parent})",
+            name = flag_info.name,
+            parent = sym.parent().trim_end_matches("::")
+        ))
```

Comment:
This is definitely very fishy. Apparently the `type_id` for an _enum type_ gets overridden with the last _enum **variant**_ within the type. That should be solved in the symbols list don't you think?

Reply:
Actual fix for this is a one-liner: https://github.com/bilelmoussaoui/gir/pull/3 - you can drop this workaround again :grin: